### PR TITLE
Lra 577 lsa ride share survey faculty interface

### PR DESCRIPTION
**Pull request by britaumich**
This PR gives faculty (the current_user.uniqname) access to surveys that were created with that uniqname.
The index page for surveys:  http://localhost:3000/faculty_surveys/faculty_index (In the header, the user will see only this link.)
This link http://localhost:3000/faculty_surveys/survey/1 (1 is the survey_id) has list of questions with rich_text fields for answers. 
The FacultySurvey model has uniqname, term_is, and unit_id fields.
Every FacultySurvey record has_many :config_questions

The Survey module takes care of creating a survey page, saving the answers, creating a program from those answers, and validating the answers. 

To test:

- login as yourself 
- click on SuperAdmin in the header and turn on Faculty Survey preference for any Unit. 
- (you can create a Test Unit with test-unit-rideshare-admins ldap group - then you can test the application with um171712 user as the Test Unit admin)
- create a faculty survey with this uniqname um174887
- login as um174887
- I hope you'll see the Faculy Survey link in the header
- click on Questions link
- try to save the survey without any answers; only with the first two, etc

When the survey is saved you'll see the link to the created program and a button to Add Sites. The button is doing nothing for now. Editing a program by a faculty is a subject for the next pull request.

### CODE REVIEW

**Comment by britaumich** [comment]
Converted this PR to a draft because realized that another one should be submitted before this one is merged.

File: app/models/survey.rb [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    else
+      survey = @faculty_survey.config_questions.order(:id)
+    end
+    return survey
+  end
+
+  def update_answers(params)
+    result = { 'success' => true, 'note' => '' }
+    course = false
+    # answers have rich_text format and params are something like this:
```
Comment (rsmoke) [inline app/models/survey.rb:40]: delete commented code

**Review by rsmoke — COMMENTED** [review]
Delete the comments in the survey model.

File: app/models/survey.rb (hunk 1 above)
Comment (britaumich) [reply app/models/survey.rb:40]: @rsmoke This is not the code, these are comments about the input format - to make it easier to understand why in the following code I had to use split, strip_tags, strip, what is 'item', etc.
```  
if p[0].split("_").first == "item"
    question_number = p[0].split("_").last
    question_id = p[0].split("_").second
    if question_number == '1' 
        if strip_tags(p[1]).strip == '' 
...
```

But if the code is understandable without comments I can delete them. Please let me know if you insist on deleting the comments.

**Comment by rsmoke** [comment]
I do not insist

-

Rick Smoke, Application Architect

For assistance please submit a request at:
https://teamdynamix.umich.edu/TDClient/47/LSAPortal/Requests/TicketRequests/NewForm?ID=lMiqIZLcXJ0_&RequestorType=ServiceOffering

College of Literature, Science, and the Arts | University of Michigan
LSA Technology Services | 301 East Liberty Street, Ann Arbor, MI 48109

**Review by rsmoke — APPROVED** [review]

**Pull request merged**